rest/oidc_api.go

```diff
@@ -237,6 +237,13 @@ func (h *handler) getOIDCCallbackURL() string {
 		base.Warnf("Can't calculate OIDC callback URL without DB in path.")
 		return ""
 	} else {
-		return fmt.Sprintf("%s://%s/%s/%s", scheme, h.rq.Host, dbName, "_oidc_callback")
+		callbackURL := fmt.Sprintf("%s://%s/%s/_oidc_callback", scheme, h.rq.Host, dbName)
+		if !isDefault && providerName != "" {
+			if callbackURL, err := auth.SetURLQueryParam(callbackURL, auth.OIDCAuthProvider, providerName); err != nil {
+				base.Warnf("Failed set provider to OpenID Connect Callback URL: %s", callbackURL)
```

Comment:
Keep this and the other error message in `NewDatabaseContext` consistent, and include underlying error:

```suggestion
				base.Warnf("Failed to add provider %q to OIDC callback URL: %v", base.UD(providerName), err)
```

Reply:
Added changes to track underlying error